osu.Game/Screens/Ranking/ReplayDownloadButton.cs

```diff
@@ -83,6 +83,10 @@ private void load(OsuGame? game, ScoreModelDownloader scoreDownloader)
 
             Score.BindValueChanged(score =>
             {
+                // An export may be pending from the last score.
+                // Reset this to meet user expectations (a new score which has just been switched to shouldn't export)
+                State.ValueChanged -= exportWhenReady;
```

Comment:
Note that this isn't required for the majority of cases, but still good to have as a correctness measure.